src/Akka.Streams.Kafka/Stages/Consumers/Abstract/BaseSingleSourceLogic.cs

```diff
@@ -123,11 +128,18 @@ protected virtual void MessageHandling((IActorRef, object) args)
                 
                 case Status.Failure failure:
                     var exception = failure.Cause;
-                    if (_decider(exception) == Directive.Stop)
+                    var cause = exception is KafkaException ke ? ke.Error.Reason : exception.Message; 
+                    var directive = _decider(exception); 
+                    if (directive == Directive.Stop)
```

Comment:
What happens if we get a `Directive.Restart`, which is the default?

Reply:
(default for supervision strategies in general)

Reply:
Resume and Restart is pretty much the same at this time. Since the SourceActor is hosed every time it encounters an error, we basically have to restart the stream

Reply:
ie, clear the stage buffer and recreate a new KafkaConsumerActor

Reply:
Oh, i think i forgot something...

Reply:
Added code to check if the decider asked the stage to resume or restart when a serialization error happened